pkg/storaged/utils.js

```diff
@@ -771,8 +771,37 @@ export function find_children_for_mount_point(client, mount_point, self) {
     return children;
 }
 
-export function get_fstab_config_with_client(client, block, also_child_config) {
-    let config = block.Configuration.find(c => c[0] == "fstab");
+export function get_fstab_config_with_client(client, block, also_child_config, subvol) {
+    function match(c) {
+        let default_subvol;
+        if (subvol) {
+            const btrfs_volume = client.blocks_fsys_btrfs[block.path];
+            const default_subvolid = client.uuids_btrfs_default_subvol[btrfs_volume.data.uuid];
+            if (default_subvolid === subvol.id) {
+                default_subvol = subvol.pathname;
```

Comment:
I would change this to `is_default_subvol = true;`.

 The `default_subvol` variable is not going to be the pathname of the default subvol in general here.

Reply:
Further down below we have:
```
            if (subvol.pathname == default_subvol &&
```
Which can never be true, thinking about it, it's about the case whee you don't specify a `subvol=`. However I am not sure if that is relevant, this piece of code honestly confuses me a lot.

Reply:
Honestly, that whole duplicate conditional is weird. Refactored it.